app/mm-redux/utils/channel_utils.ts

```diff
@@ -43,10 +43,10 @@ export function buildDisplayableChannelListWithUnreadSection(usersState: UsersSt
     const locale = getUserLocale(currentUserId, profiles);
     const missingDirectChannels = createMissingDirectChannels(currentUserId, myChannels, myPreferences);
     const channels = buildChannels(usersState, myChannels, missingDirectChannels, teammateNameDisplay, locale);
-    const unreadChannels = [...buildChannelsWithMentions(channels, myMembers, locale), ...buildUnreadChannels(channels, myMembers, locale)];
+    const unreadChannels = [...buildChannelsWithMentions(channels, myMembers, locale), ...buildUnreadChannels(channels, myMembers, locale, collapsedThreadsEnabled)];
 
     // collapsedThreadsEnabled is set to "false" as we are filtering not just based on root posts
-    const notUnreadChannels = channels.filter((channel: Channel) => !isUnreadChannel(myMembers, channel, false));
+    const notUnreadChannels = channels.filter((channel: Channel) => !isUnreadChannel(myMembers, channel, collapsedThreadsEnabled));
```

Comment:
I think this should be `false` because we want channels excluding the root condition check.

Line 46 (your fix) solves one problem and this solves a different one.

Reply:
cc @ashishbhate

Reply:
@amyblais I'm chatting with Anurag to figure out why this change is required. Will update soon.

Reply:
@ashishbhate I did test with different scenarios and it works .. I got confused with the logic.